templater.py

```diff
@@ -39,39 +39,69 @@ def __len__(self):
         return len(self.__keys)
 
 
+def no_duplicates_constructor(loader, node, deep=False):
+  """Check for duplicate keys."""
+  mapping = {}
+  for key_node, value_node in node.value:
+    key = loader.construct_object(key_node, deep=deep)
+    if key in mapping:
+      msg = "Duplicate key {0} (overwrite existing value '{1}' with new value '{2}'"
+      msg = msg.format(key, mapping[key], value_node)
+      raise yaml.YAMLError(msg)
+    value = loader.construct_object(value_node, deep=deep)
+    mapping[key] = value
+  return loader.construct_mapping(node, deep)
+
+
+def construct_mapping(loader, node):
+  loader.flatten_mapping(node)
+  return object_pairs_hook(loader.construct_pairs(node))
+
+
+class DupCheckLoader(yaml.Loader):
+  """Local class to prevent pollution of global yaml.Loader."""
+  pass
+
+
 def load_config(config_file):
     """
     Loads the yaml config file specified in 'config_file' and process it as yaml.
     Returns processed config
     """
     config = []
+    DupCheckLoader.add_constructor(yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG,
```

Comment:
This `add_contructor` call should probably go in the global namespace rather than in a function.